### PR TITLE
FF-232  FF-233  Update Script 

### PR DESCRIPTION
**Pull request by qvalentin**

### CODE REVIEW

File: config.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 app_port=80
 db_user=root
 db_name=filefighter
-use_stable_versions=true
-db_password=
+use_stable_versions=false
+db_password=BXEyIBFxVoOwpNyJgGWFuNmQRrsYPOiKBYlpunIHtXOzoOrQAR
```
Comment (open-schnick) [inline config.cfg:5]: bruh.

**Comment by open-schnick** [comment]
Looks fine to me, didnt test it.

File: lib/update.sh [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    echo "Config for db_password not found, using defaults."
+    db_password="$(read config.cfg.defaults db_password)"
+  fi
+
+  # Finished Config:
+  echo "Finished reading config. starting update..."
+
+  # Check if (default) password was empty.
+  if ! [[ $db_password ]]; then
+    echo "Password was not set, please install first."
```
Comment (open-schnick) [inline lib/update.sh:55]: oks set the password in your config `configPath` or run 'ffigther install' again.

File: lib/update.sh [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+   ffupdateStable
+  else
+    echo "Updating latest versions. Be aware that minor bugs could occur. Please report found bugs: filefigther@t-online.de."
+    ffupdateLatest
+  fi
+
+
+
+
+
```
Comment (open-schnick) [inline lib/update.sh:71]: what are these spaces xd

File: lib/update.sh [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+
+ffupdateLatest(){
+
+echo "Warning! Updating latest version, this is not recommended"
+
+if ! type "regctl" > /dev/null; then
+ echo "regctl not found! Install it from here https://github.com/regclient/regclient/releases"
+ exit 1
```
Comment (open-schnick) [inline lib/update.sh:132]: maybe we should start listing deps we need.

- docker
- wget / curl
- regctl

File: lib/update.sh [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    --name $restname filefighter/rest:latest >/dev/null 2>&1
+
+  echo "Finished downloading. Starting the updated container..."
+  docker start $restname
+
+  echo ""
+fi
+
+
+
```
Comment (open-schnick) [inline lib/update.sh:191]: xd

File: lib/update.sh [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+}
+
```
Comment (open-schnick) [inline lib/update.sh:193]: xDDDD

**Review by open-schnick — CHANGES_REQUESTED** [review]
Maybe Clean it up a bit. But does look good. Did you test it?

**Comment by qvalentin** [comment]
You can test the latest update by installing the stable version, then changing the config to use latrst and then running update. Run it twice, the second time it should not update.

Stable update can be tested when we reales new tags in the two repos.

**Review by open-schnick — APPROVED** [review]
I fixed minor spelling and bumped up version to 1.5. Also tested it and looks fine.

**Pull request merged**